subprojects/packagefiles/aklomp-base64/meson.build

```diff
@@ -0,0 +1,182 @@
+project(
+  'aklomp-base64',
+  'c',
+  version: '0.5.2',
+  license: 'BSD-2-Clause',
+  meson_version: '>=0.59.0',
+  default_options: ['warning_level=3'],
+)
+
+cc = meson.get_compiler('c')
+tests_opt = get_option('tests').disable_auto_if(meson.is_subproject())
+
+c_args = ['-DBASE64_EXPORTS']
+config = configuration_data()
+
+if get_option('default_library') == 'static'
+  c_args += ['-DBASE64_STATIC_DEFINE']
+endif
+
+neon32_supported = host_machine.cpu_family() == 'arm' and cc.has_argument(
+  '-mfpu=neon',
+)
+
+is_x86 = host_machine.cpu_family() in ['x86', 'x86_64']
+config.set10('BASE64_WITH_NEON32', neon32_supported)
+config.set10('BASE64_WITH_NEON64', host_machine.cpu_family() == 'aarch64')
+
+present_flags = {}
+arch_flags = {}
+
+if cc.get_id() == 'msvc'
+  arch_flags += {
+    'SSSE3': [],
+    'SSE41': [],
+    'SSE42': [],
+    'AVX': ['/arch:AVX'],
+    'AVX2': ['/arch:AVX2'],
+    'AVX512': ['/arch:AVX512'],
+  }
+
+  present_flags += {
+    'SSSE3': true,
+    'SSE41': true,
+    'SSE42': true,
+  }
+
+  config.set10('BASE64_WITH_SSSE3', is_x86)
+  config.set10('BASE64_WITH_SSE41', is_x86)
+  config.set10('BASE64_WITH_SSE42', is_x86)
+else
+  arch_flags += {
+    'SSSE3': ['-mssse3'],
+    'SSE41': ['-msse4.1'],
+    'SSE42': ['-msse4.2'],
+    'AVX': ['-mavx'],
+    'AVX2': ['-mavx2'],
+    'AVX512': ['-mavx512vl', '-mavx512vbmi'],
+  }
+endif
+
+foreach feature, flags : arch_flags
+  present = is_x86 and cc.has_multi_arguments(flags)
+  config.set10('BASE64_WITH_' + feature, present)
+  present_flags += {
```

Comment:
You could simplify by removing `present_flags` entirely and setting `arch_flags[feature] = []` if not present.